app/Http/Controllers/Api/Dashboard/MenuController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Http\Controllers\Api\Dashboard;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Auth;
+use App\Entities\Menu;
+class MenuController extends Controller
+{
+    
+    public function index(){
+        $roles = Auth::User()->roles->pluck('id');
+        $menus = Menu::whereHas('roles', function($q) use ($roles) {
```

Comment:
try to do this by using a Criteria or Eloquent Scopes

Reply:
```
 public function scopeConstraintRoles($q, $roles){
        
        return $q->whereHas('roles', function($q) use ($roles) {
            $q->whereIn('role_id', $roles);
        })
        ->get()
        ->map(function($v){
            return [
                'name' => $v->name,
                'path' => $v->path
            ];
        })->values()->all();
    }
```
I transferred it scope...